examples/atari/dqn/train_dqn.py

```diff
@@ -169,22 +135,45 @@ def phi(x):
         eval_stats = experiments.eval_performance(
             env=eval_env,
             agent=agent,
-            n_steps=None,
-            n_episodes=eval_n_episodes)
-        print('n_runs: {} mean: {} median: {} stdev {}'.format(
-            eval_n_episodes, eval_stats['mean'], eval_stats['median'],
+            n_steps=args.eval_n_steps,
+            n_episodes=None)
+        print('n_episodes: {} mean: {} median: {} stdev {}'.format(
+            eval_stats['episodes'],
+            eval_stats['mean'],
+            eval_stats['median'],
             eval_stats['stdev']))
     else:
         experiments.train_agent_with_evaluation(
             agent=agent, env=env, steps=args.steps,
-            eval_n_steps=eval_n_steps,
-            eval_n_episodes=eval_n_episodes,
+            eval_n_steps=args.eval_n_steps,
+            eval_n_episodes=None,
             eval_interval=args.eval_interval,
             outdir=args.outdir,
-            save_best_so_far_agent=False,
+            save_best_so_far_agent=True,
             eval_env=eval_env,
         )
 
+        dir_of_best_network = os.path.join(args.outdir, "best")
+        agent.load(dir_of_best_network)
+
+        # Change seed for final evaluation
+        final_seed = max(train_seed, test_seed) - \
+            min(train_seed, test_seed) - 1
+        eval_env.seed(int(final_seed))
+        # run 30 evaluation episodes, each capped at 5 mins of play
+        stats = experiments.evaluator.eval_performance(
+            env=eval_env,
+            agent=agent,
+            n_steps=None,
+            n_episodes=args.n_best_episodes,
+            max_episode_len=4500,
+            logger=None)
+        stats["episodes"] = args.n_best_episodes
+        with open(os.path.join(args.outdir, 'bestscores.txt'), 'a+') as f:
+            print("The results of the best scoring network:\n", file=f)
+            for stat in stats:
+                print(str(stat) + ":" + str(stats[stat]), file=f)
```

Comment:
I think it is better to use an easy-to-parse format like json to save `stats`. It would be still helpful to print a human-readable text like this to stdout though.